build-system/pr-check/remote-tests.js

```diff
@@ -73,17 +78,20 @@ async function main() {
     if (buildTargets.has('RUNTIME') ||
         buildTargets.has('BUILD_SYSTEM') ||
         buildTargets.has('UNIT_TEST')) {
-      timedExecOrDie('gulp test --unit --nobuild --saucelabs_lite');
+      exitCodes.push(
+          timedExec('gulp test --unit --nobuild --saucelabs_lite'));
     }
 
     if (buildTargets.has('RUNTIME') ||
         buildTargets.has('BUILD_SYSTEM') ||
         buildTargets.has('INTEGRATION_TEST')) {
-      timedExecOrDie('gulp test --integration --nobuild --saucelabs');
+      exitCodes.push(
+          timedExec('gulp test --integration --nobuild --saucelabs'));
     }
     stopSauceConnect(FILENAME);
   }
 
+  process.exitCode = exitCodes.some(code => code != 0) ? 1 : 0;
```

Comment:
I see that you're trying to run all tests instead of exiting early if one of them fails.

One problem I see with doing so is that if a PR has bad code that causes a lot of tests to time out, the job will now spend way longer waiting for all tests to time out before failing the job.

At the minimum, I'd move this change to its own PR so that it can be cleanly reverted if what I suspect does come true.

Reply:
Makes sense! Reverted.